stable/node-feature-discovery/templates/daemonset.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "node-feature-discovery.fullname" . }}
+  labels:
+{{ include "node-feature-discovery.labels" . | indent 4 }}
+spec:
+  selector:
+    matchLabels:
+{{ include "node-feature-discovery.matchLabels" . | indent 6 }}
+  template:
+    metadata:
+      labels:
+{{ include "node-feature-discovery.labels" . | indent 8 }}
+    spec:
+      serviceAccountName: {{ template "node-feature-discovery.serviceAccountName" . }}
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      containers:
+        - env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          name: nfd-master
+          command:
+            - "nfd-master"
+          args:
+            - "--extra-label-ns=nvidia.com"
+          securityContext:
+{{- toYaml .Values.securityContext | nindent 12 }}
+          resources:
+{{ toYaml .Values.resources | indent 12 }}
+        - env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
```

Comment:
The node name (and maybe other?) Env cars which are common to master and workder can be defined in values and reused for the 2 container specs , which would also make it easier to add stuff and override by users

Reply:
there is logic from the official repository... so let's leave it as is and I definitely sure that will never need to change/extend https://github.com/kubernetes-sigs/node-feature-discovery/search?q=Getenv&unscoped_q=Getenv

Reply:
I don't understand how the code is related to the fact it's duplicated and not extendible/overridable....or why you sure no one will want to add an env var ever... Here's an example of how to take environment from values
https://github.com/v3io/helm-charts/blob/development/stable/jupyter/templates/jupyter-deployment.yaml#L117
Then you add this block with the NODE_NAME var defined in values file.
Wherever env vars are affecting application (==always) we usually have to play with them / add to them sooner or later